src/main/java/com/stripe/net/RequestOptions.java

```diff
@@ -155,13 +174,19 @@ public RequestOptionsBuilder clearClientId() {
       return this;
     }
 
-    public RequestOptionsBuilder setStripeVersion(String stripeVersion) {
-      this.stripeVersion = normalizeStripeVersion(stripeVersion);
-      return this;
-    }
-
-    public RequestOptionsBuilder clearStripeVersion() {
-      this.stripeVersion = null;
+    /**
+     * Do not use this except for in API where JSON response is not fully deserialize into
+     * to explicit Stripe resources -- it intends to make request on behalf of others, makes
+     * request with their API version, and simply passes the raw values along. Example for this is
+     * {@link com.stripe.model.EphemeralKey#create(Map, RequestOptions)}.
+     * Setting this value in typical scenario will result in deserialization error
+     * as the model classes has schema according to the pinned {@link Stripe#API_VERSION} and not
+     * the {@code stripeVersionOnBehalfOf}
+     * @param stripeVersionOnBehalfOf stripe version of the client to make request on behalf of
+     * @return option builder
+     */
+    public RequestOptionsBuilder setStripeVersionOnBehalfOf(String stripeVersionOnBehalfOf) {
```

Comment:
Not too sure about the naming here — can we just call it `setStripeVersion`? We can specify in the comments that it'll override the built-in version.

Reply:
@brandur, I'm wondering if i should make the name different here. I'm afraid that merchants who are using pre-pinning version still have `setStripeVersion` invocations, and once they migrate to pinned-version will not realize the semantic difference here. 
Making it explicitly different here, and remove the old method will at least ensure that they set up the integration correctly.

Would naming like `overrideStripeVersion` would be a better alternative?

Reply:
Oh oops — I didn't read carefully enough and didn't realize that `setStripeVersion` could already be called for `RequestOptions` before.

Yeah, I was actually going to suggest using "override" in the name as an alternative before, so totally okay with that. I'd suggest `stripeVersionOverride` instead of `overrideStripeVersion` though since it sorts more nicely.

Reply:
acked!